tests/phpunit/tests/query/cacheResults.php

```diff
@@ -827,6 +845,70 @@ public function test_query_cache_delete_meta() {
 		$this->assertNotSame( $query1->found_posts, $query2->found_posts );
 	}
 
+
+	/**
+	 * @ticket 56802
+	 */
+	public function test_query_cache_like_meta() {
+		global $wpdb;
+		$p1 = self::$posts[0];
+
+		$args        = array(
+			'cache_results' => true,
+			'fields'        => 'ids',
+			'meta_query'    => array(
+				array(
+					'key'     => 'color',
+					'value'   => '00',
+					'compare' => 'LIKE',
+				),
+			),
+		);
+		$query1      = new WP_Query();
+		$posts1      = $query1->query( $args );
+		$num_queries = get_num_queries();
+
+		// Force placeholder to be different.
+		$wpdb->placeholder_escape( true );
+
+		$query2 = new WP_Query();
+		$posts2 = $query2->query( $args );
+
+		$this->assertSame( $posts1, $posts2 );
```

Comment:
Per [Writing PHP Tests - Using Assertions](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions), these assertions should have a `message` parameter to explain what went wrong if the assertion fails. However, since the rest of the test file doesn't have them, it may be preferred to handle the whole file in another ticket.

🔢 Applies the to test method below as well.